census/tftransformestimator/README.md

```diff
@@ -0,0 +1,348 @@
+
+python preprocess.py --train-data-file $TRAIN_FILE_LOCAL \
+                     --test-data-file $EVAL_FILE_LOCAL \
+                     --root-train-data-out train \
+                     --root-test-data-out test \
+                     --working-dir $TFT_OUTPUT_DIR_LOCAL \
+                     --temp_location ${TFT_OUTPUT_DIR_LOCAL}/tmp
+
+```
+
+## Preprocessing with Cloud Dataflow
+
+```
+gcloud auth application-default login
+
+export TFT_OUTPUT_DIR=gs://<my-bucket>/path/to/my/jobs/
+export REGION=<your-gcp-region>
+export PROJECT=<your-gcp-project>
+export JOB_NAME=census-tft
+
+python preprocess.py --train-data-file $TRAIN_FILE \
+                     --test-data-file $EVAL_FILE \
+                     --root-train-data-out train \
+                     --root-test-data-out test \
+                     --working-dir $TFT_OUTPUT_DIR \
+                     --runner DataflowRunner \
+                     --requirements_file requirements_dataflow.txt \
+                     --region $REGION \
+                     --project $PROJECT \
+                     --job_name $JOB_NAME
+```
+
+
+# Single Node Training
+Single node training runs TensorFlow code on a single instance. You can run the exact
+same code locally and on Cloud ML Engine.
+
+## How to run the code
+You can run the code either as a stand-alone python program or using `gcloud`.
+See options below:
+
+### Using local python
+Run the code on your local machine:
+
+```
+export TRAIN_STEPS=1000
+DATE=`date '+%Y%m%d_%H%M%S'`
+export OUTPUT_DIR=census_$DATE
+rm -rf $OUTPUT_DIR
+```
+
+### Model location reuse
+Its worth calling out that unless you want to reuse the old model output dir,
+model location should be a new location so that old model doesn't conflict with new
+one.
+
+```
+python -m trainer.task --tft-working-dir $TFT_OUTPUT_DIR_LOCAL \
+                       --train-filebase train \
+                       --eval-filebase test \
+                       --train-steps $TRAIN_STEPS \
+                       --job-dir $OUTPUT_DIR
+
+```
+
+### Using gcloud local
+Run the code on your local machine using `gcloud`. This allows you to mock
+running it on the cloud:
+
+```
+export TRAIN_STEPS=1000
+DATE=`date '+%Y%m%d_%H%M%S'`
+export OUTPUT_DIR=census_$DATE
+rm -rf $OUTPUT_DIR
+```
+
+```
+gcloud ml-engine local train --module-name trainer.task \
+                             --package-path trainer \
+                             -- \
+                             --tft-working-dir $TFT_OUTPUT_DIR_LOCAL \
+                             --train-filebase train \
+                             --eval-filebase test \
+                             --train-steps $TRAIN_STEPS \
+                             --job-dir $OUTPUT_DIR
+
+```
+
+### Using Cloud ML Engine
+Run the code on Cloud ML Engine using `gcloud`. Note how `--job-dir` comes
+before `--` while training on the cloud and this is so that we can have
+different trial runs during Hyperparameter tuning.
+
```

Comment:
Consider adding link to a further explanation of these training flags in the docs:

"See more information about [training job arguments](https://cloud.google.com/ml-engine/docs/tensorflow/training-jobs#submitting_the_job)."

Reply:
Done